### ISSUE TITLE
(Client) OpenCV Application Development

### ISSUE THREAD

**Issue by abbottwhitley** [cu-ecen-5013/final-project-abbottwhitley#14]
Develop OpenCV application to provide video stream to client requests. 

- [x] Main Thread: Generate video stream and write image data to a global structure
- [x] Setup socket connection to support client requests for video stream 
- [x] Each client request to be handled by a separate thread 

DoD: Ability to view multiple video stream from separate client requests.

**Comment by abbottwhitley** [comment]
Blocked by https://github.com/cu-ecen-5013/final-project-abbottwhitley/issues/8

**Comment by abbottwhitley** [comment]
Server C++ source code updated as outlined by the issue subtask checkboxes (see above). C++ client works as expected but python client doesn't quite assemble the image properly. See image below (python client on bottom left, C++ client on top right)

Reference this [commit.](https://github.com/cu-ecen-5013/opencv-monitoring-system/commit/57a6d55d0c90e52616b2c29f753049d696e475f6)

![image](https://user-images.githubusercontent.com/2907462/115058963-12f78000-9ea3-11eb-999b-78a1290dc7ab.png)

**Comment by abbottwhitley** [comment]
Resolved issue by pausing socket `send` command on server side. Will trouble shoot further in another issue focusing on fine tuning the system. Assuming there will be other issues with the data feed from the actual RPI.

**Comment by abbottwhitley** [comment]
Tested setup on RPI4. Ran OpenCV server application from RPI4. Connected to RPI4 using two client connections to test operation of separate threads serving video stream. Image below shows C++ OpenCV client connection and python HTML client connection. 

![image](https://user-images.githubusercontent.com/2907462/115093296-10168280-9ed7-11eb-9c45-aa78eedfe13d.png)


[opencv-monitoring-system commit associated with this issue](https://github.com/cu-ecen-5013/opencv-monitoring-system/commit/5ccdc7d63e808e0bcfbe9ce06d6c87811eaf550d) 

[Main Buildroot repo commit associated with this issue](https://github.com/cu-ecen-5013/final-project-abbottwhitley/commit/b660e1f997e44deafdc80ee1b26d775fab1500e5)
